src/uipath/_services/attachments_service.py

```diff
@@ -0,0 +1,780 @@
+                
+        return attachment_key
+
+    @traced(name="attachments_upload", run_type="uipath")
+    async def upload_async(
+        self,
+        *,
+        name: str,
+        source_path: str,
+        folder_key: Optional[str] = None,
+        folder_path: Optional[str] = None,
+    ) -> uuid.UUID:
+        """Upload a file to UiPath as an attachment asynchronously.
+
+        This method asynchronously uploads a file to UiPath and makes it available as an attachment.
+
+        Args:
+            name (str): The name of the attachment file.
+            source_path (str): The local path of the file to upload.
+            folder_key (Optional[str]): The key of the folder. Override the default one set in the SDK config.
+            folder_path (Optional[str]): The path of the folder. Override the default one set in the SDK config.
+
+        Returns:
+            uuid.UUID: The UUID of the created attachment.
+
+        Raises:
+            Exception: If the upload fails.
+
+        Examples:
+            ```python
+            import asyncio
+            from uipath import UiPath
+
+            client = UiPath()
+
+            async def main():
+                attachment_key = await client.attachments.upload_async(
+                    name="my-document.pdf",
+                    source_path="path/to/local/document.pdf",
+                )
+                print(f"Uploaded attachment with key: {attachment_key}")
+            ```
+        """
+        spec = self._create_attachment_and_retrieve_upload_uri_spec(
+            name=name,
+            folder_key=folder_key,
+            folder_path=folder_path,
+        )
+
+        result = (
+            await self.request_async(
+                spec.method,
+                url=spec.endpoint,
+                params=spec.params,
+                headers=spec.headers,
+                json=spec.json,
+            )
+        ).json()
+
+        # Get the ID from the response and convert to UUID
+        attachment_key = uuid.UUID(result["Id"])
+
+        upload_uri = result["BlobFileAccess"]["Uri"]
+        headers = {
+            key: value
+            for key, value in zip(
+                result["BlobFileAccess"]["Headers"]["Keys"], result["BlobFileAccess"]["Headers"]["Values"], strict=False
+            )
+        }
+
+        with open(source_path, "rb") as file:
+            if result["BlobFileAccess"]["RequiresAuth"]:
+                await self.request_async(
+                    "PUT", upload_uri, headers=headers, files={"file": file}
+                )
+            else:
+                request("PUT", upload_uri, headers=headers, files={"file": file})
+                
+        return attachment_key
+
+    @traced(
+        name="attachments_upload_from_memory",
+        run_type="uipath",
+        input_processor=_upload_attachment_input_processor,
+    )
+    def upload_from_memory(
```

Comment:
we should have only one `upload()` method with 2 variants: `upload(path=)` or `upload(content=)`